pkg/ottl/expression_test.go

```diff
@@ -676,6 +677,44 @@ func Test_StandardStringGetter(t *testing.T) {
 	}
 }
 
+func Test_StandardFunctionGetter(t *testing.T) {
+	tests := []struct {
+		name             string
+		getter           StandardFunctionGetter[any]
+		want             ExprFunc[any]
+		valid            bool
+		expectedErrorMsg string
+	}{
+		{
+			name: "function type",
+			getter: StandardFunctionGetter[any]{
+				Getter: func(ctx context.Context, tCtx any) (interface{}, error) {
+					return ExprFunc[any](func(ctx context.Context, tCtx any) (interface{}, error) {
+						return "str", nil
+					}), nil
+				},
+			},
+			want:  func(context.Context, any) (interface{}, error) { return "str", nil },
+			valid: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			function, err := tt.getter.Get(context.Background(), nil)
+			funcVal, _ := function(context.Background(), nil)
+			if tt.valid {
+				assert.NoError(t, err)
+				isEqual := reflect.TypeOf(funcVal) == reflect.TypeOf(tt.want)
```

Comment:
Can we update the test to assert that the value of `funcVal` is exactly what we expect (in this case "str")?

Reply:
I updated the test with a string assertion